internal/project/validation.go

```diff
@@ -21,8 +21,8 @@ func init() {
 }
 
 // GetValidatedModel validates an incoming proto request and returns a project model
-func GetValidatedModel(ctx context.Context, req *metrov1.Project) (*Model, error) {
-	if req.Name == "" {
+func GetValidatedModel(ctx context.Context, req *metrov1.Project, allowEmptyName bool) (*Model, error) {
```

Comment:
this seems hackish, why do we need project with empty name

Reply:
delete call takes projectID in the URL path, where project name can be empty. But when creating a project, name cannot be empty. But in both cases, I am preparing a model to create or delete the entity and hence this way

Reply:
should be have different validators for different operations than combining into one method

Reply:
done